rbac_api_with_domains_test.go

```diff
@@ -267,3 +268,22 @@ func TestGetAllDomains(t *testing.T) {
 
 	testGetAllDomains(t, e, []string{"domain1", "domain2"})
 }
+
+func TestDomainHierarchical(t *testing.T) {
+	e, _ := NewEnforcer("examples/rbac_with_domains_hierarchical_model.conf", "examples/rbac_with_domains_hierarchical_policy.csv")
+	origin := e.GetNamedRoleManager("g").(*defaultrolemanager.RoleManager)
+	dest := e.GetNamedRoleManager("g2").(*defaultrolemanager.RoleManager)
+	origin.SetDomainHierarchyManager(dest)
```

Comment:
Should we add a helper function, so that you can enable the hierarchy without digging up the RoleManager? Like `EnableDomainHierarchy` which would set DomainHierachy to `g` with `g2` by default.

Reply:
@Sefriol 
> I think it's not possible to define g2 in the format of child, parent. At least in nodeJS, it provided incorrect results, but maybe golang has a different RoleManager definition.
> You can try to replicate the test case I did for nodeJS. I think it provides quite good coverage on all the possible use cases. Especially the 4 domain structure is essential.
> I have tried to comment on why certain decisions were made in the pull request and a little bit about the thought behind the hierarchy model.

Maybe you should use ``conf`` and ``csv`` files so that we can replicate each other's use cases

Reply:
It might be good, but I don't have time to do it right now